docs/reference.rst

```diff
@@ -53,6 +53,18 @@ Format specification
   # runtime environment variables.
   env:
     - <var name>: <value>
+
+  # List of architecture-specific ROCKs to be built.
+  # Supported architectures are: amd64, arm64, arm, i386, ppc64le, riscv64 and s390x.
+  platforms:
+    # If matching a valid architecture name, it must be the same as "build-for".
+    <entry>:
```

Comment:
I think I'm confused on the definition of the entities here: can you please clarify?
- `Supported architectures` refers to the `build-on` and `build-for` values, right? That is, the `<arch>` entries;
- If this is true, what exactly *is* each `<entry>` platform? Can you tell me a use-case for a platform that would not match a valid architecture name?

Reply:
yes, you're right.

The extra bit of information you can add to your reasoning is: "if <entry> matches a `supported architecture`, then its name is taken into account for the business logic.

here's an  example: https://github.com/canonical/rockcraft/pull/52/files#diff-31c6bafe300f15ab74a13b7445a1e801360baa32ce1b99c0c025c1edb27bd67f